src/components/modal/modal_header_title.tsx

```diff
@@ -20,9 +20,14 @@ export const EuiModalHeaderTitle: EuiModalHeaderTitleProps = ({
   ...rest
 }) => {
   const classes = classnames('euiModalHeader__title', className);
+  const renderChildren = () =>
+    React.Children.map(children, (child) => {
+      if (typeof child === 'string') return <h1>{child}</h1>;
+      return child;
+    });
   return (
     <div className={classes} {...rest}>
-      {children}
+      {renderChildren()}
```

Comment:
I'm not 100% on the `typeof children`, but I think this should be more robust. I'd try to avoid `map`ing children if possible in the edge case that someone passes an array/map of strings, in which case you'd end up with multiple `<h1>`s (another accessibility violation)

```suggestion
  const childrenWithHeading = useMemo(() =>
    if (typeof children === 'string') return <h1>{children}</h1>;
    return children;
  }, [children]);

  return (
    <div className={classes} {...rest}>
      {childrenWithHeading}
```

Reply:
Good call @constancecchen. I refactored using your recommended approach, and found what I think is a better way to reason about `children` being a string than **typeof**. Source article here: https://overreacted.io/why-do-react-elements-have-typeof-property/